lib/hx/hx_v2.ml

```diff
@@ -0,0 +1,447 @@
+open Base
+open Tyxml_html
+
+module Verb = struct
+  type t =
+    [ `Delete of Uri.t
+    | `Get of Uri.t
+    | `Patch of Uri.t
+    | `Post of Uri.t
+    | `Put of Uri.t
+    ]
+
+  let delete_attr uri = Uri.to_string uri |> Unsafe.string_attrib "hx-delete"
+  let get_attr uri = uri |> Uri.to_string |> Unsafe.string_attrib "hx-get"
+  let patch_attr uri = uri |> Uri.to_string |> Unsafe.string_attrib "hx-patch"
+  let post_attr uri = uri |> Uri.to_string |> Unsafe.string_attrib "hx-post"
+  let put_attr uri = uri |> Uri.to_string |> Unsafe.string_attrib "hx-put"
+
+  let to_attr = function
+    | `Delete uri -> delete_attr uri
+    | `Get uri -> get_attr uri
+    | `Patch uri -> patch_attr uri
+    | `Post uri -> post_attr uri
+    | `Put uri -> put_attr uri
+  ;;
+
+  let uri = function
+    | `Delete uri | `Get uri | `Patch uri | `Post uri | `Put uri -> uri
+  ;;
+
+  let uri_to_string verb = verb |> uri |> Uri.to_string
+
+  let verb_to_string = function
+    | `Delete _ -> "DELETE"
+    | `Get _ -> "GET"
+    | `Patch _ -> "PATCH"
+    | `Post _ -> "POST"
+    | `Put _ -> "PUT"
+  ;;
+
+  let to_string verb =
+    Fmt.str "%s %s" (verb_to_string verb) (uri_to_string verb)
+  ;;
+end
+
+module Css = struct
+  type t =
+    [ `Added
+      (** Applied to a new piece of content before it is swapped, removed after it is settled. *)
+    | `Indicator
+      (** A dynamically generated class that will toggle visible (opacity:1) when a htmx-request class is present *)
+    | `Request
+      (** Applied to either the element or the element specified with hx-indicator while a request is ongoing *)
+    | `Settling
+      (** Applied to a target after content is swapped, removed after it is settled. The duration can be modified via hx-swap. *)
+    | `Swapping
+      (** Applied to a target before any content is swapped, removed after it is swapped. The duration can be modified via hx-swap. *)
+    ]
+
+  let to_string = function
+    | `Added -> "hx-added"
+    | `Indicator -> "hx-indicator"
+    | `Request -> "hx-request"
+    | `Settling -> "hx-settling"
+    | `Swapping -> "hx-swapping"
+  ;;
```

Comment:
so tempted to make a ppx for hx_show 😆

Reply:
Honestly just a general ppx that takes any variant/polymorphic variant and converts it to a snake case string would be amazing lol